### DISCUSSION TITLE
rich markdown editor

### DISCUSSION THREAD

**Discussion by mickael-kerjean** [outline/rich-markdown-editor#221] [General] [answered]
Hey guys,

I discovered this project from hacker news and got to say this is quite an awesome job that you did.
I've been working on integrating it with my [open source project](https://github.com/mickael-kerjean/filestash). Unfortunately, my enthusiasm made me not read the entire README carefully and I missed the: `This project is not attempting to be an all-purpose Markdown editor`

 so far I did a [few changes to the code base](https://github.com/mickael-kerjean/rich-markdown-editor/commits/master) that goes along the line of an all purpose markdown editor:
1. support for line break (when doing a Shift+Enter)
2. support for front matter syntax with both YAML and TOML that are used as config by static website generators. The current markdownit parser is having isssues with this at the moment and consider the last parameter as a h2 title. Also none of the plugin I've tried did work so I had to create a small plugin for markdownit
3. fix a js error that would generate a `Uncaught ReferenceError: regeneratorRuntime is not defined` when clicking on a link
4. removal of smart text features as it doesn't align with the general purpose markdown editor
5. remove the '/' character at the beginning of each empty paragraph to align again with a general purpose markdown editor

My intentions are to use this as the basis of:

1. a full fledge markdown editor to edit static website using either Jekyll of Hugo directly from git. The current editor is based on codemirror (https://demo.filestash.app/login?next=/view/_posts/2018-05-31-release-note-v0.1.md#type=git&repo=https://github.com/mickael-kerjean/filestash-website) and this would improve it quite a bit

2. a full fledge org-mode editor to create wikis based on org-mode. At the moment, the org mode editor is made using codemirror( see: https://demo.filestash.app/login?next=/view/Documents/getting_started.org#type=webdav&url=https://webdav.filestash.app)

Ultimately, I've open this issue as I'm not sure what to do with the changes I've made, keep them into the fork? create a pull request? What do you prefer?

**Comment by tommoor** [comment, marked as answer]
Hi Mickael, thanks for the enthusiasm – it does sound like the changes you've made are largely quite specific to your project so I'd encourage you to keep them on a fork, with the exception being any fixes for bugs that exist over here in the main project, we'd really appreciate PR's addressing those.

**Comment by mickael-kerjean** [comment]
will do that then, thanks you!
